python/pyspark/sql/pandas/serializers.py

```diff
@@ -155,19 +157,14 @@ def _create_batch(self, series):
         from pyspark.sql.pandas.types import _check_series_convert_timestamps_internal, \
             _convert_dict_to_map_items
         from pandas.api.types import is_categorical_dtype
-        # Make input conform to [(series1, type1), (series2, type2), ...]
-        if not isinstance(series, (list, tuple)) or \
-                (len(series) == 2 and isinstance(series[1], (pa.DataType, DataType))):
-            series = [series]
-        series = ((s, None) if not isinstance(s, (list, tuple)) else s for s in series)
 
-        def create_array(s, dt: DataType, t: pa.DataType):
-            mask = s.isnull()
-            if isinstance(dt, UserDefinedType):
-                s = s.apply(dt.serialize)
-            elif isinstance(dt, ArrayType) and isinstance(dt.elementType, UserDefinedType):
-                udt = dt.elementType
-                s = s.apply(lambda x: [udt.serialize(f) for f in x])
+        def create_array(s, t: pa.DataType, dt: Optional[DataType] = None):
+            if dt is not None:
+                if isinstance(dt, UserDefinedType):
```

Comment:
this part together with `pa.StructArray.from_pandas` is too tricky I need more backgrounds of Arrow and Pandas.